### PR TITLE
Christmas Tree by Xiaochen Xue and Jing Yu

### PR DESCRIPTION
**Pull request by LetUsWorship**
Please approve!!!!

### CODE REVIEW

File: 01-python/infrastructure/tree.py [hunk 1]
```diff
@@ -1,3 +1,24 @@
...
-    print("tree not found")
+def draw_tree(size=3):
+    Triangle = ""
+    Triangle2 = ""
+    Triangle3 = ""
+    for i in range (size):
+        Triangle += ("*"*((i*4)+1)).center((((size+4)*4))," ")
+        Triangle += "\n"
+    Triangle += ("|"*((i*4)+3)).center((((size+4)*4))," ")
+    print(Triangle)
```
Comment (gallettilance) [inline 01-python/infrastructure/tree.py]: when size=0 you are printing some new lines but you should not be printing anything if size is 0

File: 01-python/infrastructure/tree.py [hunk 2]
```diff
@@ -1,3 +1,28 @@
-def draw_tree():
-    print("tree not found")
+def draw_tree(size=3):
+
+    if size <= 0:
+        raise ValueError("Invalid Size for a Tree")
```
Comment (gallettilance) [inline 01-python/infrastructure/tree.py]: I don't think 0 is an invalid size. I think you should just return if you get a size of 0.

Comment (gallettilance) [reply 01-python/infrastructure/tree.py]: what do you think?

File: 01-python/infrastructure/tree.py [hunk 3]
```diff
@@ -1,3 +1,28 @@
-def draw_tree():
-    print("tree not found")
+def draw_tree(size=3):
```
Comment (gallettilance) [inline 01-python/infrastructure/tree.py]: Please add [docstrings](https://www.python.org/dev/peps/pep-0257/) to your function that specifies the parameters and return values as in [this example](https://www.geeksforgeeks.org/python-docstrings/)

File: 01-python/infrastructure/tree.py (hunk 2 above)
Comment (LetUsWorship) [reply 01-python/infrastructure/tree.py]: I think this is a good idea! (making changes now)

File: 01-python/infrastructure/tree.py (hunk 3 above)
Comment (LetUsWorship) [reply 01-python/infrastructure/tree.py]: good idea sir!

Comment (gallettilance) [reply 01-python/infrastructure/tree.py]: did you forget about this?

**Pull request closed without merging**